circle.yml

```diff
@@ -1,26 +1,40 @@
 machine:
   environment:
-    AWS_ACCESS_KEY_ID: a
-    AWS_SECRET_ACCESS_KEY: b
+    AWS_ACCESS_KEY_ID: 1
+    AWS_SECRET_ACCESS_KEY: 1
+    GO15VENDOREXPERIMENT: 1
+    GOPATH: $HOME/gocode
+    PATH: $GOPATH/bin:$PATH
+    ZENPATH: $HOME/gocode/src/github.com/zencoder
+    SRC_DIR: $ZENPATH/fastly-tokens
 checkout:
   post:
-    - rm -rf ~/.go_workspace/src/github.com/zencoder
-    - mkdir -p ~/.go_workspace/src/github.com/zencoder
-    - cp -R ~/fastly-tokens ~/.go_workspace/src/github.com/zencoder/fastly-tokens
+    - mv "$ZENPATH" /tmp || echo
+    - mkdir -p "$ZENPATH"
+    - cp -R ~/fastly-tokens "$SRC_DIR"
+    - mkdir -p ~/download_cache
 dependencies:
+  cache_directories:
+    - "~/download_cache"
+    - "~/gocode/bin"
   pre:
     - sudo apt-get remove --purge golang
     - sudo rm -rf /usr/local/go/
-    - mkdir /tmp/go && wget 'https://storage.googleapis.com/golang/go1.5.1.linux-amd64.tar.gz' -O /tmp/go/go1.5.1.linux-amd64.tar.gz
-    - sudo tar -xzf /tmp/go/go1.5.1.linux-amd64.tar.gz -C /usr/local
+    - cd ~/download_cache && wget --no-clobber 'https://storage.googleapis.com/golang/go1.5.3.linux-amd64.tar.gz'
+    - cd ~/download_cache && sudo tar -xzf go1.5.3.linux-amd64.tar.gz -C /usr/local
   override:
-    - go get github.com/tools/godep
-    - go get golang.org/x/tools/cmd/cover
-    - go get github.com/mattn/goveralls
-    - go get github.com/modocache/gover
+    - type cover     || go get golang.org/x/tools/cmd/cover
+    - type goveralls || go get github.com/mattn/goveralls
+    - type gover     || go get github.com/modocache/gover
+  post:
+    - mv ~/.go_workspace /tmp || echo
```

Comment:
No more `rm`ing these unused dirs? Why the `|| echo`? Seems like a simple `;` would do it, but nbd either way.